packages/ui/src/primitives/tailwind/Tooltip/index.tsx

```diff
@@ -23,96 +23,185 @@ All portions of the code written by the Infinite Reality Engine team are Copyrig
 Infinite Reality Engine. All Rights Reserved.
 */
 
-import React, { ReactNode, useEffect, useRef, useState } from 'react'
+import React, { ReactNode, useEffect, useImperativeHandle, useRef, useState } from 'react'
+import ReactDOM from 'react-dom'
 import './tooltip.css'
 
-export interface TooltipProps {
+export interface BaseTooltipProps {
   title?: string
   content: ReactNode
   children: ReactNode
   position?: 'auto' | 'top' | 'bottom' | 'left' | 'right'
 }
 
-function Tooltip({ title, content, children, position = 'auto' }: TooltipProps) {
+export interface ControlledProps {
+  isControlled: true
+  onMouseEnter: () => boolean
+  onMouseLeave: () => boolean
+}
+
+export interface UncontrolledProps {
+  isControlled?: false
+}
+
+export type TooltipProps = BaseTooltipProps & (ControlledProps | UncontrolledProps)
+
+export interface TooltipRef {
+  showTooltip: () => void
+  hideTooltip: () => void
+}
+
+function Tooltip(
```

Comment:
lets add a jsdoc to explain the `ref` provides those 2 functions

```suggestion
/**
 * Ref:

 * Provides an imperative handle with showTooltip and hideTooltip methods to manually control the tooltip visibility.
*/
```

Reply:
done